### ISSUE TITLE
Load dashboard with default values

### ISSUE THREAD

**Issue by rajkaran** [fcorti/adf-pentaho-project#3]
Can we set field values default or being calculated on page load and dashboard loads with these param values.

Once dashboard load is completed user can change parameter accordingly.

**Comment by fcorti** [comment]
It should be doable... but before I need to understand how to do it into the `requirejs` function.
Assuming the following command, I need to figure out how to setup the variables in the function.

```
require([
  'dash!/public/plugin-samples/pentaho-cdf-dd/pentaho-cdf-dd-require/cde_sample1.wcdf'
], function(SampleDash) {
  var sampleDash = new SampleDash("content1");
  sampleDash.render();
  });
});
```

To trigger a change in the dashboard the command is following one:
```
  sampleDash.on("cdf productLine:fireChange", function (evt) { ... });
```

If someone have any idea of the command to setup a variable using the `sampleDash` object, please share to save some effort. ;-)

**Comment by fcorti** [comment]
Maybe this can help:
https://help.pentaho.com/Documentation/8.0/Developer_Center/CDF_API

**Comment by fcorti** [comment]
To be experimented:
https://help.pentaho.com/Documentation/7.0/0R0/cdf._Dashboard.Bootstrap#setParameter

in 

```
dashboard.on('cdf cdf:postInit', function(e) {
//I could want to keep a count of all the times my dashboard was visited, for statistical purposes
addOneDashboardInitToStatistic();
});
```

**Comment by rajkaran** [comment]
we need to add something like `drilldown.fireChange("fromDatetime", '2018-01-05 00:00:00')` for every parameter.

This can be achieved tweaking `getJsCodeForDashboardDependingOnHtmlElement` function

```
private getJsCodeForDashboardDependingOnHtmlElement(
    htmlId:string,
    params: string[],
    masterHtmlElementIds: string[]):string {

    var jsCode = ", function(Dashboard) { ";
    jsCode += "var currentDashboard = new Dashboard(\"" + htmlId + "\"); ";
    jsCode += "currentDashboard.render(); ";
    for (let i in masterHtmlElementIds) {
      jsCode += "var htmlElement" + i + " = document.getElementById(\"" + masterHtmlElementIds[i] + "\"); ";

      jsCode += "currentDashboard.fireChange(\"" + params[i] + "\", htmlElement"+ i +".value);";

      jsCode += "htmlElement" + i + ".addEventListener(\"change\", function() { currentDashboard.fireChange(\"" + params[i] + "\", this.value); }); ";
    }
    jsCode += "} ";

    return jsCode;
  }
```
besides this change, will require a boolean variable to activate ` jsCode += "currentDashboard.fireChange(\"" + params[i] + "\", htmlElement"+ i +".value);";` line conditionally.

I am working onto add `setDefaults` variable.

**Comment by fcorti** [comment]
Ok, included in the project.
Released in npm with version 1.0.4.
Well done @rajkaran!

**Comment by rajkaran** [comment]
great!!   Thank you
